folly/compression/Compression.cpp

```diff
@@ -70,7 +71,7 @@ namespace io {
 
 Codec::Codec(
     CodecType type,
-    Optional<int> level,
+    const Optional<int>& level,
```

Comment:
Passing `Optional<T>` by value should be fine when passing `T` by value is fine. Here and elsewhere.